spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2285,15 +2261,31 @@ object QueryPlanSerde extends Logging with CometExprShim {
         }
 
       case op =>
-        // Emit warning if:
-        //  1. it is not Spark shuffle operator, which is handled separately
-        //  2. it is not a Comet operator
-        if (!op.nodeName.contains("Comet") && !op.isInstanceOf[ShuffleExchangeExec]) {
-          val msg = s"unsupported Spark operator: ${op.nodeName}"
-          emitWarning(msg)
-          withInfo(op, msg)
+        opSerdeMap.get(op.getClass) match {
+          case Some(handler) =>
+            handler.enabledConfig match {
+              case Some(enabledConfig) =>
+                if (!enabledConfig.get(op.conf)) {
+                  withInfo(
+                    op,
+                    s"Native support for operator ${op.getClass.getSimpleName} is disabled. " +
+                      s"Set ${enabledConfig.key}=true to enable it.")
+                  return None
+                }
+              case _ =>
+            }
+            handler.asInstanceOf[CometOperatorSerde[SparkPlan]].convert(op, builder, childOp: _*)
```

Comment:
if we know handler is only instanceOf `CometOperatorSerde[SparkPlan]` should we have the strong typed it in `opSerdeMap` ?